src/main/kotlin/org/rust/ide/template/postfix/stringBasedPostfixTemplates.kt

```diff
@@ -61,3 +66,39 @@ class MatchPostfixTemplate(provider: RsPostfixTemplateProvider) :
         }
     }
 }
+
+abstract class ForPostfixTemplate(provider: RsPostfixTemplateProvider, pred: (RsExpr) -> Boolean, example: String) :
+    StringBasedPostfixTemplate("iter", example, RsTopMostInScopeSelector(pred), provider) {
+
+    override fun getElementToRemove(expr: PsiElement): PsiElement = expr
+}
+
+private val RsExpr.implementsIter: Boolean
+    get() {
+        val items = StdKnownItems.relativeTo(this)
+        val implLookup = ImplLookup(project, items)
+        val iterTrait = items.findIteratorTrait() ?: return false
+        return implLookup.canSelect(TraitRef(type, iterTrait.withDefaultSubst()))
+    }
+
+private val RsExpr.implementsIntoIter: Boolean
+    get() {
+        val items = StdKnownItems.relativeTo(this)
+        val implLookup = ImplLookup(project, items)
+        val intoIterTrait = items.findIntoIteratorTrait() ?: return false
+        return implLookup.canSelect(TraitRef(type, intoIterTrait.withDefaultSubst()))
+    }
```

Comment:
`implementsIter` and `implementsIntoIter` look very similar
Maybe extract common code?